### PR TITLE
Add "Deposit funds" button to allow merchants to request a deposit of their available balance

### LINKED ISSUE #4485: [Manual deposits] Merchants can request a deposit of their available balance via the "Deposit funds" button
Part of the manual deposits project - pdjTHR-1DS-p2

### Blocked by
- https://github.com/Automattic/woocommerce-payments/issues/4464
- 2265-gh-Automattic/woocommerce-payments-server

Please ensure the above issues are completed before working on this issue.

### Description

The custom deposit schedule project will let merchants change their deposit schedule to "Manual". In doing so, they can request an on-demand deposit of their available balance via the "Deposit funds" button located on the **Payments → Overview** page in WCPay Client.

Screen Shot 2022-10-27 at 10 23 18 am

In this issue, we need to implement the "Deposit funds" button workflow.

### Acceptance criteria

- A "Deposit funds" button appears on the **Payments → Overview** page when the merchant's deposit schedule is set to "Automatic - weekly", "Automatic - monthly" or "Manual".
- The "Deposit funds" button is enabled if:
	- The merchant has deposits enabled (i.e. the `deposits_blocked` flag is not `0`)
	- **and**
	- The merchant's available balance is greater than the minimum deposit amount for the deposit currency (e.g. $5.00).
- When the "Deposit funds" button is clicked, a modal appears that asks the merchant to confirm the deposit request.
- When the deposit request is confirmed, a request is sent to WCPay Server and is processed by the endpoint handler (see 2265-gh-Automattic/woocommerce-payments-server).
- If successful:
	- A payout is initiated and the balances component is updated to reflect the new state.
	- A success notice is visible and contains the text: `Deposit for %%deposit-amount%% in transit.` The notice includes a "View details" link which links to the deposit details page for the new deposit.

**Post-release:** Add user docs for this feature to https://woocommerce.com/document/payments/

### Designs

paJDYF-6a4-p2

### Testing instructions

TBD

>To send the funds from a test transaction directly to your available balance, use the test cards in this section. Other test cards send funds from a successful payment to your pending balance.

https://stripe.com/docs/testing#available-balance

### Additional context
- Project thread: pdjTHR-1DS-p2
- PRD: pdjTHR-1DP-p2

### PR DESCRIPTION
**Pull request by Jinksi**
Fixes #4485

#### Changes proposed in this Pull Request

* This PR adds a new "Deposit funds" button on the Payments → Overview screen for eligible merchants.
* Clicking on this button will display a confirmation modal where the merchant can either confirm or cancel the request. 
* Confirming the request via the modal will send a `POST` request to  `/wc/v3/payments/deposits` with the `standard` type.
* Once a response is received, relevant deposit data stores are invalidated and re-fetched, resulting in an updated view of the deposit balance status.
* If the `POST` request was successful, the deposits balance status should now show the expected deposit amount in the "next deposit" box.

#### Merchant eligibility
* The account's deposit schedule is set to `Automatic - weekly`, `Automatic - monthly` or `Manual`.
* The account is not soft or hard-blocked.

If the merchant has met the criteria above, but has no available deposit balance (non-pending), the "Deposit now" button will be visible but disabled.

#### Testing instructions

**Preparation**

* Switch to the `fix/2477-test-mode-for-deposit-schedule-2` branch on your local WCPay Server.
* Follow the "Preparation" instructions listed here → https://github.com/Automattic/woocommerce-payments-server/pull/2495
* In WCPay server, empty your `wcpay_test_deposits_cache` table.

**Reonboard**

* Reonboard on your store using a United States address.
* After completing onboarding, navigate to **Payments → Overview**.
* Observe the balances box:

Screen Shot 2022-10-26 at 3 38 27 pm

* Observe that the "Deposit funds" button is not present. This is expected because the merchant has not completed the new account waiting period.

**First order**

* Create a $1 USD product on your store and publish it.
* Add the product to your cart and navigate to checkout.
* Observe the checkout total is $1 USD.
* Complete checkout using the card number `4000000000000077`. This card bypasses the standard pending period and adds the funds directly into your available balance. This is useful for testing deposits.
* The new account waiting period will commence.
* Navigate to **Payments → Overview**. 
* Observe the balances box:

Screen Shot 2022-10-26 at 3 55 36 pm

* Observe that the "Deposit funds" button is still not present. This is because the new account waiting period has not been completed yet.
* Wait 5 minutes.
* The new account waiting period is now completed.
* Navigate to WCPay Dev Tools and clear the account cache contents.
* Navigate to **Payments → Overview**.
* Observe the balances box:

Screen Shot 2022-10-26 at 3 55 36 pm

* Observe that the "Deposit funds" button is still not present. This is because the merchant's deposit schedule is set to `Daily` and as such the merchant is not eligible for manual deposits.

**Change deposit schedule**

* Navigate to **Payments → Settings**.
* Scroll down to the "Deposits" section.
* Change the deposit schedule from `Daily` to `Monthly` and save the changes.
* Navigate to **Payments → Overview**.
* Observe the balances box:

Screen Shot 2022-10-26 at 4 04 27 pm

* Observe that a disabled "Deposit funds" button is now present.

**Balance is below the minimum deposit amount**

* Hover the "Deposit funds" button.

Screen Shot 2022-10-27 at 9 55 03 am

* Observe a tooltip appears with the text:
> Deposits require a minimum available balance of $5.00.

* Purchase a different product from your store using the card number `4000000000000077` to bring the available balance above the $5 minimum.
* Navigate to **Payments → Overview**.
* Observe the "Deposit funds" button is now enabled and a tooltip **does not** appear when hovering the button.

Screen Shot 2022-10-27 at 10 23 18 am

**Request a manual deposit**

* Switch to the `add/2597-transaction-ids-to-overview-all` branch on your local WCPay Server.
* Navigate to **Payments → Overview**.
* Click the "Deposit funds" button.
* Observe a confirmation modal appears:

Screen Shot 2022-10-27 at 10 27 48 am

* Click the "Submit deposit" button.
* Observe the button changes to a "busy" state while the request is being processed.
* Observe the modal closes a toast notification appears to confirm that the deposit request was successful:

Screen Shot 2022-10-27 at 10 30 17 am

* Observe the balances component is updated to reflect the manual deposit:

Screen Shot 2022-10-27 at 10 31 20 am

* Observe the "Pending balance", "Next deposit" and "Available balance" components display `$0.00` while the "Last deposit" component displays the last deposit amount, e.g. `$17.85`.

**Cannot request multiple manual deposits within a 24 hour period**

* Navigate to **Payments → Overview**.
* Hover the "Deposit funds" button.

Screen Shot 2022-10-27 at 10 40 36 am

* Observe a tooltip appears with the text:
> Deposits are available once per day. Your next deposit can be created on October 28, 2022 at 11:00AM.

**Instant deposits**

* Create a $1000 USD product on your store and publish it.
* Add the product to your cart and navigate to checkout.
* Observe the checkout total is $1000 USD.
* Complete checkout using the card number `4000000000000077`.
* Navigate to **Payments → Overview**.
* Click the "Edit details" link in the "Account details" box.
* You will be navigated to the Stripe merchant dashboard. (Use text message code 000000 if asked)
* Click the person icon in the top right corner.
* Click the button in the "Payout details" section.
* Edit the existing bank account by clicking the pencil icon.
* Switch the bank account to a debit card and save the changes.
* Your store is now eligible for instant deposits.
* In WCPay server I had to manually add the following code to the `get_or_add()` function in `server/wp-content/rest-api-plugins/endpoints/wcpay/utils/class-cache.php` to clear the cache:

```php
wp_cache_delete( $key, self::CACHE_GROUP );
```
* With this temporary code in place, navigate to WCPay Dev Tools and clear the account cache contents. The `instant_deposits_eligible` key should now be `true`.
* Navigate to **Payments → Overview**.
* Observe, the deposit button is now labelled "Instant deposit".

Screen Shot 2022-10-27 at 2 28 32 pm

**Note:** Ignore the incorrect amount in the "Available balance" box. There is currently a bug in the WCPay Server branch `add/2597-transaction-ids-to-overview-all`.

* Hover the "Instant deposit" button, observe a tooltip appears with the text:
> Deposits are available once per day. Your next deposit can be created on October 28, 2022 at 11:00AM.

**Instant deposits - balance is below the minimum deposit amount**

* Complete the steps listed in the *Preparation* section above.
* Reonboard on your store using a United States address.
* Purchase a product from your store that is less than $50 USD using the card number `4000000000000077`.
* Wait 5 minutes for the new account waiting period to complete.
* In WCPay Server, manually add `return true;` to the top of the `calculate_instant_deposit_eligibility()` function to force allow instant deposits. This prevents us from needing to satisfy the $1000 TPV requirement.
* Navigate to **Payments → Overview**.
* Click the "Edit details" link in the "Account details" box.
* You will be navigated to the Stripe merchant dashboard.
* Click the person icon in the top right corner.
* Click the button in the "Payout details" section.
* Edit the existing bank account by clicking the pencil icon.
* Switch the bank account to a debit card and save the changes.
* Navigate to WCPay Dev Tools and clear the account cache contents. The `instant_deposits_eligible` key should now be `true`.
* Navigate to **Payments → Overview**.
* Observe the disabled "Instant deposit" button.
* Hover the "Instant deposit" button and observe a tooltip appears with the text:
> Instant deposits require a minimum available balance of $50.00.

Screen Shot 2022-10-27 at 3 05 50 pm

**Instant deposits - happy path**

* Purchase some additional products from your store using the card number `4000000000000077` to bring your available balance above $50 USD.
* Navigate to **Payments → Overview**.
* Observe the "Instant deposit" button is now enabled and a tooltip **does not** appear when hovering the button.
* Click the "Instant deposit" button.
* Observe a confirmation modal appears:

Screen Shot 2022-10-27 at 3 14 03 pm

**Note:** I have not changed the contents of this modal in this PR. This is all existing content / functionality.

* Click the "Deposit now" button.
* Observe the button changes to a "busy" state while the request is being processed.
* Observe the modal closes a toast notification appears to confirm that the instant deposit request was successful:

Screen Shot 2022-10-27 at 3 15 22 pm

* Observe the balances component is updated to reflect the instant deposit:

Screen Shot 2022-10-27 at 3 16 48 pm

* Observe the "Pending balance", "Next deposit" and "Available balance" components display `$0.00` while the "Last deposit" component displays the last deposit amount, e.g. `$68.27`.

**Instant deposits - cannot request multiple manual deposits within a 24 hour period**

* Navigate to **Payments → Overview**.
* Hover the "Instant deposit" button, observe a tooltip appears with the text:
> Deposits are available once per day. Your next deposit can be created on October 28, 2022 at 3:15PM.

Screen Shot 2022-10-27 at 3 54 00 pm

**Multiple deposit currencies**

* Complete the steps listed in the *Preparation* section above.
* Reonboard on your store using a Canadian address, during Stripe KYC add both a Canadian bank account and a US bank account. 
* After completing onboarding, navigate to **Payments → Overview**.
* Observe the balances box:

Screen Shot 2022-10-27 at 4 38 19 pm

* Observe that the "Deposit funds" button is not present. This is expected because the merchant has not completed the new account waiting period.

**Enable multi-currency**

* Navigate to **Payments → Settings**.
* Scroll to the bottom of the page and click the "Advanced settings" button.
* Ensure the "Enable Multi-Currency" checkbox is enabled and save the changes.
* Navigate to **WooCommerce → Settings → Multi-currency**.
* Ensure both $USD and $CAD are added as enabled currencies and save the changes.

**Multiple deposit currencies - balance is below the minimum deposit amount**

* Navigate to the front end of your store. Ensure the currency switcher (located in the site's sidebar) is set to $CAD.
* Add the $1 product we created earlier to your cart and navigate to checkout.
* Observe the checkout total is some small amount in $CAD.
* Complete checkout using the card number `4000000000000077`.
* The new account waiting period will commence.
* Navigate to **Payments → Overview**. 
* Observe the balances box:

Screen Shot 2022-10-27 at 4 53 51 pm

* Observe that the "Deposit funds" button is still not present. This is because the new account waiting period has not been completed yet.
* Wait 5 minutes.
* The new account waiting period is now completed.
* Navigate to WCPay Dev Tools and clear the account cache contents.
* Navigate to **Payments → Overview**.
* Observe the balances box:

Screen Shot 2022-10-27 at 4 53 51 pm

* Observe that the "Deposit funds" button is still not present. This is because the merchant's deposit schedule is set to `Daily` and as such the merchant is not eligible for manual deposits.
* Navigate to **Payments → Settings**.
* Scroll down to the "Deposits" section.
* Change the deposit schedule from `Daily` to `Monthly` and save the changes.
* Navigate to **Payments → Overview**.
* Observe the balances box:

Screen Shot 2022-10-27 at 4 59 05 pm

* Observe that a disabled "Deposit funds" button is now present in both the CAD and USD balance boxes.
* Hover the "Deposit funds" button in the CAD balance box.

Screen Shot 2022-10-27 at 5 01 11 pm

* Observe a tooltip appears with the text:
> Deposits require a minimum available balance of $5.00.

**Multiple deposit currencies - standard CAD deposit**

* Purchase another product from your store in $CAD using the card number `4000000000000077` to bring the available balance above the $5 minimum.
* Navigate to **Payments → Overview**.
* Observe the balances box:

Screen Shot 2022-10-28 at 1 53 07 pm

* Observe that the CAD "Deposit funds" button is enabled while the USD "Deposit funds" button remains disabled.
* Hover over the USD "Deposit funds" button.

Screen Shot 2022-10-28 at 1 57 03 pm

* Observe a tooltip appears with the text:
> Deposits require a minimum available balance of $5.00.
* Click the CAD "Deposit funds" button.
* Observe a confirmation modal appears:

Screen Shot 2022-10-28 at 1 55 35 pm

* Click the "Submit deposit" button.
* Observe the button changes to a "busy" state while the request is being processed.
* Observe the modal closes a toast notification appears to confirm that the deposit request was successful.
* Observe the CAD balances component is updated to reflect the manual deposit:

Screen Shot 2022-10-28 at 1 58 54 pm

* Observe the "Pending balance", "Next deposit" and "Available balance" components display `$0.00` while the "Last deposit" component displays the last deposit amount, e.g. `$46.31`.

**Multiple deposit currencies - cannot request multiple manual deposits within a 24-hour period (CAD)**

* Hover over the CAD "Deposit funds" button.

Screen Shot 2022-10-28 at 2 43 41 pm

* Observe a tooltip appears with the text:
> Deposits are available once per day. Your next deposit can be created on October 29, 2022 at 11:00AM.

**Multiple deposit currencies - standard USD deposit**

* Purchase another product from your store in $USD using the card number `4000000000000077` to bring the available balance above the $5 minimum.
* Navigate to **Payments → Overview**.
* Observe the balances box:

Screen Shot 2022-10-28 at 2 49 50 pm

* Observe the USD "Deposit funds" button is now enabled while the CAD "Deposit funds" button remains disabled.
* Click the USD "Deposit funds" button.
* Observe a confirmation modal appears:

Screen Shot 2022-10-28 at 2 51 33 pm

* Click the "Submit deposit" button.
* Observe the button changes to a "busy" state while the request is being processed.
* Observe the modal closes a toast notification appears to confirm that the deposit request was successful.
* Observe the USD balances component is updated to reflect the manual deposit:

Screen Shot 2022-10-28 at 2 52 26 pm

* Observe the "Pending balance", "Next deposit" and "Available balance" components display `$0.00` while the "Last deposit" component displays the last deposit amount, e.g. `$52.20`.

**Multiple deposit currencies - cannot request multiple manual deposits within a 24-hour period (USD)**

* Hover over the USD "Deposit funds" button.

Screen Shot 2022-10-28 at 2 53 47 pm

* Observe a tooltip appears with the text:
> Deposits are available once per day. Your next deposit can be created on October 29, 2022 at 11:00AM.

#### Todo

- [x] Determine what params are required for server POST request as per 2415-gh-Automattic/woocommerce-payments-server
- [x] Implement POST request in `submitStandardDeposit`: https://github.com/Automattic/woocommerce-payments/blob/3b9c83f4892d272b5f1051743eb97472dffeb131/client/data/deposits/actions.js#L169
- [ ] Test error handling
- [ ] Consider adding more unit tests or e2e tests including unhappy paths
- [ ] Write testing instructions including unhappy paths

#### Notes

* The POST request to `/wc/v3/payments/deposits` is almost directly forwarded to WCPay-server: https://github.com/Automattic/woocommerce-payments/blob/884c7fdf6f997f4c82c3c1d808784b507e09d297/includes/admin/class-wc-rest-payments-deposits-controller.php#L180-L184

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: client/components/deposits-information/test/index.js
Comment (Jinksi) [inline client/components/deposits-information/test/index.js:80]: I noticed this `useInstantDeposit` `isLoading` prop should be named `inProgress`, as per naming in https://github.com/Automattic/woocommerce-payments/blob/884c7fdf6f997f4c82c3c1d808784b507e09d297/client/data/deposits/hooks.ts#L187

File: client/components/standard-deposit-button/style.scss
Comment (Jinksi) [inline client/components/standard-deposit-button/style.scss]: This is to add a small margin between the modal footer buttons. I'm not sure if this is the intended use of these variables but I've seen them used in various places and it works in this case.

https://github.com/Automattic/woocommerce-payments/blob/3b9c83f4892d272b5f1051743eb97472dffeb131/client/stylesheets/abstracts/_variables.scss#L5

File: client/components/standard-deposit-button/modal.tsx
Comment (Jinksi) [inline client/components/standard-deposit-button/modal.tsx]: I've used the explicit currency format here (e.g. `$14.15 AUD`), which diverges from the design but is more informative for the merchant IMO.

File: client/data/deposits/hooks.ts
Comment (Jinksi) [inline client/data/deposits/hooks.ts:190]: This is very similar to the `useInstantDeposit` hook. I'm tempted to merge them into a single hook. However, I decided that it is best to stick to the "one function for one task" paradigm, rather than merge and use a param e.g. `depositType: 'instant' | 'standard'`.

File: client/data/deposits/actions.js
Comment (Jinksi) [inline client/data/deposits/actions.js:165]: This is very similar to `submitInstantDeposit`. This could also be merged into one, but looking at existing actions, having an explicit action for each task seems to be preferred.

**Comment by Jinksi** [comment]
I've put development on hold for this issue since manual deposits work will continue as part of an upcoming project: pdjTHR-1c8-p2#comment-2009

**Comment by james-allan** [comment]
I've picked this up for now. I'll shift to other team work next week but I'll push this forward today and tomorrow.

**Comment by Jinksi** [comment]
@james-allan FYI I had trouble naming things for this feature 🫠

It started as the "Deposit Now" button before turning into the "Manual Deposit" button and then into the "Standard Deposit" button. 😵‍💫

IIRC, "Standard Deposit" lined up with Stripe terminology.

**Comment by Jinksi** [comment]
@aprea Going through the testing steps, I've hit a hurdle on step **Request a manual deposit**:

* Switch to the wcpay-server branch `add/2597-transaction-ids-to-overview-all`
* Click the **Submit deposit** button
* `/wc/v3/payments/deposits` responds status `500` with this error message:
  `Error: Deposits are not allowed until the account has completed the waiting period`

I figure that this branch doesn't have the logic required to allow the shorter test account waiting period: _WCPay Dev Tools → Account cache contents_ `'completed_waiting_period' => false`.

As a workaround, I'll see if I have an existing account that has passed the `completed_waiting_period` with which I can test.

**Comment by aprea** [comment]
> @aprea Going through the testing steps, I've hit a hurdle on step **Request a manual deposit**:

Ah interesting, I wonder if this is related to the `is_proxied_automattician()` issue we discussed in Slack? It's not ideal, but would you be able to add a temporary `return true;` to the top of that function to see if it gets you further?

**Review by Jinksi — COMMENTED** [review]
✅🎉 I've stepped through the testing steps provided, and all is working as expected, except for a few things I've marked below with ⚠️.

@aprea can you take a look at the ⚠️ comments/questions below?

I'll leave this as a comment and submit a code review separately.

**Testing checklist:**
- [x] Reonboard
	* Using local WCPay env with test account + sandboxed WCPay-server synced with branch  `fix/2477-test-mode-for-deposit-schedule-2`. WP 6.0.3, WC 7.1.0-dev
	* Balance box matches
- [x] First order
	* Available balance $0.67
	* Deposit Funds button doesn’t display when deposit schedule set to Daily.
- [x] Change deposit schedule
	* Change deposit schedule to Monthly, button is displayed
- [x] Balance is below the minimum deposit amount
- [x] Request a manual deposit
	* Switched to WCPay-server `add/2597-transaction-ids-to-overview-all`
	* I had to hard-code the `has_account_completed_the_new_account_waiting_period()` response to `true`
	* Balances reflect changes correctly.
- [x] Cannot request multiple manual deposits within 24h
	* Button disabled with correct tooltip content.
- [x] Instant deposits
	* 	Updated Stripe payouts to debit card.
	* Used `wp_cache_delete( $key, self::CACHE_GROUP );` to become eligible.
	* Observed instant deposit button. Disabled with `Deposits are available once per day` tooltip.
- [x] Instant deposits - balance is below the minimum deposit amount
	* Reonboarded & preparation steps completed.
	* Purchased products with balance below $50.
	* Instant deposit button displayed and disabled with correct tooltip content.
- [x] Instant deposits - happy path
	* Purchased product to bring balance over $1000, which disabled Instant Balance. 
	  
	* ⚠️ The tooltip message could be improved here to indicate that Instant Deposit is disabled because it’s *over* $1000 rather than *under* $50.
	* I’m unsure if this scenario was just an artefact of testing. Also, this may be out of the scope of this PR.
	* Upon re-onboarding, I discovered that there is also a $250 threshold that prevents instant deposits ([code ref](https://github.com/Automattic/woocommerce-payments-server/blob/47417a61ebd8b53355abc87db9b96af36b0d4cb4/server/wp-content/rest-api-plugins/endpoints/wcpay/service/class-deposit-service.php#L709)). Should this be clarified to the merchant via the tooltip?
	* Deposit processed, notification & balances correct.
- [x] Instant deposits - cannot request multiple manual deposits within a 24 hour period
	* After increasing the Available Balance above $50 again, the “instant deposit button” remains disabled with the correct tooltip.
- [x] Multiple deposit currencies
	* Onboarded with CAD & USD bank accounts
	* No deposit funds button displayed
- [x] Enable multi-currency
- [x] Multiple deposit currencies - balance is below the minimum deposit amount
	* ⚠️ The deposit funds button doesn’t display when shcedule set to Manual. Setting to Weekly or Monthly will display the button.
- [x] Multiple deposit currencies - standard CAD deposit
	* Once reaching the minimum deposit threshold, the CAD deposit now button displays while USD remains disabled.
	* CAD deposit successfully completes. Clicking on toast notification action redirects to the correct payout in Payments → Deposits.
	* Balances updated correctly.
- [x] Multiple deposit currencies - cannot request multiple manual deposits within a 24-hour period (CAD)
	* CAD Button is disabled and shows the correct tooltip.
	* USD Button shows the minimum amount tooltip.
- [x] Multiple deposit currencies - standard USD deposit
	* Topping up CAD balance + USD balance above minimum deposit threshold results in only USD button enabled. CAD disabled due to previous deposit within 24h. 
	* USD deposit successfully completes. Clicking on toast notification action redirects to the correct payout in Payments → Deposits.
	* Balances updated correctly.
- [x] Multiple deposit currencies - cannot request multiple manual deposits within a 24-hour period (USD)
	* Both buttons appear disabled.
	* ⚠️ With the next deposit date `November 3, 2022 at 12:00AM.` Viewing the Stripe dashboard, the Payout datetime was `2 Nov 2022, 4:06 am` UTC. Should the tooltip date be `November 3, 2022 at 04:06AM`?

File: client/components/deposits-information/index.tsx
Comment (Jinksi) [inline client/components/deposits-information/index.tsx:29]: I'm curious, why did you change `interface` to `type` here? I see both used throughout this repo, seemingly interchangeably, and it's not clear to me when to use each.

File: client/globals.d.ts
Comment (Jinksi) [inline client/globals.d.ts]: It looks like `delay_days` can also be sent as `null`, as per https://github.com/Automattic/woocommerce-payments/blob/57eb845ca3562595e847dcd199ba3a940f68e711/includes/class-wc-payments-account.php#L341

```suggestion
			delay_days: number | null;
```

File: client/deposits/instant-deposits/index.js
Comment (Jinksi) [inline client/deposits/instant-deposits/index.js]: It's not obvious to me that `lastManualDeposit` here refers to a manual deposit within the last 24h.

My first thought was `lastManualDeposit` = "the previous manual deposit (regardless of the last 24h)".

I'm not sure if this is worth improving, perhaps a comment would help, e.g. ("will return true if there has been a manual deposit within the last 24h").

**Comment by aprea** [comment]
> ⚠️ The tooltip message could be improved here to indicate that Instant Deposit is disabled because it’s over $1000 rather than under $50.

This is a good improvement but would require server changes as currently, the server returns `0` when the instant balance amount is above the threshold. I'm a little hesitant to introduce any additional changes this close to code freeze. Hopefully, this is something we can iterate on later.

> ⚠️ The deposit funds button doesn’t display when shcedule set to Manual. Setting to Weekly or Monthly will display the button.

This happens on the `add/2597-transaction-ids-to-overview-all` branch. It's fixed in the `fix/2477-test-mode-for-deposit-schedule-2` branch and has to do with the server returning a `daily` deposit schedule even when the actual deposit schedule is set to `manual` ([code ref](https://github.com/Automattic/woocommerce-payments-server/blob/a896f65f5fd668abf4ffdca5eb831320e89fa8b8/server/wp-content/rest-api-plugins/endpoints/wcpay/stripe/class-stripe-account.php#L349-L352))

> ⚠️ With the next deposit date `November 3, 2022 at 12:00AM.` Viewing the Stripe dashboard, the Payout datetime was `2 Nov 2022, 4:06 am` UTC. Should the tooltip date be `November 3, 2022 at 04:06AM`?

I also noticed this during testing. Though this looks like a Stripe issue (maybe this only happens in test mode?). All the deposits in my local deposits table are being recorded with a 00:00 UTC date:

Screen Shot 2022-11-02 at 4 29 54 pm

I inspected a `payout.created` webhook and it looks like they are sending us the 00:00 UTC arrival date:

Screen Shot 2022-11-02 at 4 41 42 pm

Screen Shot 2022-11-02 at 4 42 49 pm

In the client, we simply add 24 hours to the last deposit date we received from server: https://github.com/Automattic/woocommerce-payments/blob/57eb845ca3562595e847dcd199ba3a940f68e711/client/components/standard-deposit-button/index.tsx#L70-L78

File: client/deposits/instant-deposits/index.js
Comment (Jinksi) [inline client/deposits/instant-deposits/index.js]: There is an opportunity to improve this by using a destructured object param (IMO):

```suggestion
const isButtonDisabled = ( { instantBalance, lastManualDeposit } ) => {
```

**Review by Jinksi — COMMENTED** [review]
LGTM @aprea 🌟

This was a big one with lots to test, but it all looks to be working as expected. Kudos! 🙌

I've left some comments with some opportunities for minor improvements (IMO) but nothing to block this merge.

**Update:** I've just noticed that GH checks are failing since the merge from `develop` ~so we should look into that before merging~ I see the discussion on slack about this (p1667353741152329-slack-CGGCLBN58).

**Also: I can't approve my own PR.** 😝 

image

File: client/deposits/instant-deposits/index.js
Comment (aprea) [reply client/deposits/instant-deposits/index.js]: Renamed `lastManualDeposit` to `lastDayManualDeposit` for clarity in 2a3a24e.

File: client/globals.d.ts
Comment (aprea) [reply client/globals.d.ts]: Fixed in 640ca5f.

File: client/components/deposits-information/index.tsx
Comment (aprea) [reply client/components/deposits-information/index.tsx:29]: Ah, this is just personal preference. I always default to `type` over `interface` because they feel a little more predictable. I found this comment which sums up my thoughts:

>Also, an interface can be extended in several locations while a type has to be fully defined in one place. For me that's +100pts for types because when I can be sure by looking at a type def that I have the whole object, while interfaces definitions can be scattered all around, possibly in different files

Source: https://dev.to/saranshk/typescript-the-difference-between-interface-and-type-ol2#comment-1ekg4

**Comment by aprea** [comment]
>Also: I can't approve my own PR. 😝

@Jinksi I'll drop an approval on this PR on your behalf if that's cool 😀

**Review by aprea — APPROVED** [review]

File: client/components/deposits-information/index.tsx
Comment (Jinksi) [reply client/components/deposits-information/index.tsx:29]: > an interface can be extended in several locations

Nice, so I guess it's a bit like `type` = `const`, `interface` = `let`.

Also, from the [TS docs/handbook](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#differences-between-type-aliases-and-interfaces):

> Type aliases and interfaces are very similar, and in many cases **you can choose between them freely**, and TypeScript will tell you if it needs something to be the other kind of declaration

😝 This is why I haven't yet decided which to use, but I'll default to `type` from now on. Thanks for the tip!

**Comment by Jinksi** [comment]
Closing as stale

**Pull request closed without merging**
